packages/node-experimental/src/opentelemetry/spanExporter.ts

```diff
@@ -0,0 +1,301 @@
+import { SpanKind } from '@opentelemetry/api';
+import type { ExportResult } from '@opentelemetry/core';
+import { ExportResultCode } from '@opentelemetry/core';
+import type { SpanExporter } from '@opentelemetry/sdk-trace-base';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { flush } from '@sentry/core';
+import { mapOtelStatus, parseOtelSpanDescription } from '@sentry/opentelemetry-node';
+import type { DynamicSamplingContext, Span, SpanOrigin, TransactionSource } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import { OTEL_ATTR_OP, OTEL_ATTR_ORIGIN, OTEL_ATTR_PARENT_SAMPLED, OTEL_ATTR_SOURCE } from '../constants';
+import { getCurrentHub } from '../sdk/hub';
+import { NodeExperimentalScope } from '../sdk/scope';
+import type { NodeExperimentalTransaction } from '../sdk/transaction';
+import { startTransaction } from '../sdk/transaction';
+import type { OtelSpan } from '../types';
+import { convertOtelTimeToSeconds } from '../utils/convertOtelTimeToSeconds';
+import { getRequestSpanData } from '../utils/getRequestSpanData';
+import type { OtelSpanNode } from '../utils/groupOtelSpansWithParents';
+import { groupOtelSpansWithParents } from '../utils/groupOtelSpansWithParents';
+import { getOtelSpanHub, getOtelSpanMetadata, getOtelSpanScope } from './spanData';
+
+/**
+ * A Sentry-specific exporter that converts OpenTelemetry Spans to Sentry Spans & Transactions.
+ */
+export class SentrySpanExporter implements SpanExporter {
+  private _finishedSpans: OtelSpan[];
+  private _stopped: boolean;
+
+  public constructor() {
+    this._stopped = false;
+    this._finishedSpans = [];
+  }
+
+  /** @inheritDoc */
+  public export(spans: OtelSpan[], resultCallback: (result: ExportResult) => void): void {
+    if (this._stopped) {
+      return resultCallback({
+        code: ExportResultCode.FAILED,
+        error: new Error('Exporter has been stopped'),
+      });
+    }
+
+    const openSpanCount = this._finishedSpans.length;
+    const newSpanCount = spans.length;
+
+    this._finishedSpans.push(...spans);
+
+    const remainingSpans = maybeSend(this._finishedSpans);
```

Comment:
q: When would there be remaining spans and what happens with them? Is it when we have multiple concurrent root spans?

Reply:
There are two scenarios, one is "normal" and will happen, and one _should_ not happen but _may_ happen (who knows):

1. When a root span is not finished yet, all the child spans will remain in there. E.g. think one http request in a transaction may be finished and go to the exporter, while the overall transaction is still running. In this case, the http span will remain here until the root span is completed.

2. Somewhere along the way some span was dropped (for whatever reason), so the parent span of this span never gets to the exporter. This _should not_ happen, but 🤷 So in this case we'll eventually clean this up and just discard the span.

Reply:
I also added a comment to explain this a bit better!

Reply:
thanks for explaining!